Constellation/Enclave/Key.hs

```diff
@@ -21,19 +23,17 @@ newKeyPair = do
 
 loadKeyPair :: (FilePath, FilePath)
             -> IO (Either String (PublicKey, Box.SecretKey))
-loadKeyPair (pubPath, privPath) = b64TextDecodeBs <$> readFile pubPath >>= \case
-    Left err    -> return $ Left err
-    Right pubBs -> case mkPublicKey pubBs of
-        Nothing  -> return $ Left "loadKeyPair: Failed to mkPublicKey"
-        Just pub -> AE.eitherDecode' <$> readFile privPath >>= \case
-            Left err     -> return $ Left err
-            Right locked -> do
-                putStrLn ("Unlocking " ++ privPath)
-                promptingUnlock locked >>= \case
-                    Left err     -> return $ Left err
-                    Right privBs -> case S.decode privBs of
-                        Nothing   -> return $ Left "Failed to S.encode privBs"
-                        Just priv -> return $ Right (pub, priv)
+loadKeyPair (pubPath, privPath) = runEitherT $ do
+    pubBs <- EitherT $ b64TextDecodeBs <$> readFileUtf8 pubPath
+    pub <- hoistMaybe "loadKeyPair: Failed to mkPublicKey" (mkPublicKey pubBs)
+    locked <- EitherT $ AE.eitherDecode' . fromStrict <$> readFile privPath
+    liftIO $ putStrLn ("Unlocking " ++ privPath)
+    privBs <- EitherT $ promptingUnlock locked
+    (pub,) <$> hoistMaybe "Failed to S.encode privBs" (S.decode privBs)
+
+  where
+    hoistMaybe :: Monad m => e -> Maybe a -> EitherT e m a
```

Comment:
How about `maybeToEitherT`? The `T` seems important to highlight here, and @joelburget pointed out that there's a precedent for this naming with [maybeToExceptT](https://hackage.haskell.org/package/transformers-0.5.4.0/docs/Control-Monad-Trans-Maybe.html#v:maybeToExceptT).